crds/operators.coreos.com_subscriptions.yaml

```diff
@@ -524,6 +524,11 @@ spec:
                                   topologyKey:
                                     description: This pod should be co-located (affinity) or not co-located (anti-affinity) with the pods matching the labelSelector in the specified namespaces, where co-located is defined as running on a node whose value of the label with key topologyKey matches that of any node on which any of the selected pods is running. Empty topologyKey is not allowed.
                                     type: string
+                    deployment_annotations:
+                      description: DeploymentAnnotations is an unstructured key value map stored with a resource that may be set by external tools to store and retrieve arbitrary metadata.
```

Comment:
Can we just call this annotations? I think it might be more useful to propogate annotations to all the resources that get created from the installation (rbac, apiservices, etc) than just deployments? Wdyt?

Reply:
That would be a much more invasive change. It would touch code that creates/updates:
- Deployment
- Pod (via the Deployment spec)
- Namespace
- Service
- CustomResourceDefinition
- APIService
- Secret
- ConfigMap
- ServiceAccount
- Role
- RoleBinding
- ClusterRole
- ClusterRoleBinding
- MutatingWebhookConfiguration
- ValidatingWebhookConfiguration
- custom resources: PrometheusRule, ServiceMonitor, PodDisruptionBudget, PriorityClass, ...

I'm not sure if my searches through the code of controllers missed any other resources. I only need to annotate deployments. If we call this `annotations`, would you accept a PR where I implement only the deployment piece, leaving the rest for someone else to contribute later? Or would the PR have to touch logic around all these other resources before it could be merged?

Thinking about UX: everything else in SubscriptionConfig explicitly configures the Pod spec in Deployments, not any other resource. Would calling this `annotations` suggest that this too will only apply to the pods?

Reply:
@kevinrizza If you do want one `annotations` field for all resources, then what about just propagating the `annotations` field of `Subscription` instead of adding a new field to `SubscriptionConfig`?

Reply:
You're getting at why I think this change is actually a hard one for us to maintain :)

My preference would be to rename this to just `annotations` and have the implementation just apply to `deployments`, `apiservices` and `pods`. The reconciliation of those all happens in more or less the same place. If someone else comes along and needs these annotations to apply to their rbac then we can deal with that when we get there.

Reply:
Renamed.